src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,253 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int HEAD_INDEX = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size);
+        if (isEmpty()
+                || index == size) {
+            add(value);
+            return;
+        }
+        if (index == HEAD_INDEX) {
+            addToHead(new Node<>(value));
+            return;
+        }
+        addToIndex(new Node<>(value), findNode(index));
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = findNode(index);
+        return targetNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = findNode(index);
+        T tempValue = targetNode.item;
+        targetNode.item = value;
+        return tempValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> targetNode = findNode(index);
+        removeNode(targetNode);
+        return targetNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> targetNode = findNode(object);
+        if (isTail(targetNode)) {
+            return false;
+        }
```

Comment:
Because we should return false, if the list doesn't contain an object. So, in findNode(object) we search through the list for required node. If we reached tail, and tail != required nod, we return tail. And if we return tail to remove(object) method, method returns false.

Reply:
I can not understand your explanation. Better write in MA chat)